flutter/integration_test/expected_performance.dart

```diff
@@ -0,0 +1,96 @@
+const _tagTflite = 'libtflitebackend';
+const _tagPixelSo = 'libtflitepixelbackend';
+const _tagCoreml = 'libcoremlbackend';
+
+/// Windows
+// cloudbuild n2-standard-4 machine
+const _tagCloudBuildX28 = 'Intel Xeon 2.80GHz';
+const _tagCloudBuildX23 = 'Intel Xeon 2.30GHz';
+const _tagRyzen5600 = 'AMD Ryzen 5 5600X 6-Core';
+
+const _tagPixel5 = 'Pixel 5';
+const _tagPixel6 = 'Pixel 6';
+
+const _tagIphone12mini = 'iPhone13,1';
+const _tagIosSimMBP2019 = 'x86_64';
+
+const Map<String, Map<String, double>> _imageClassification = {
+  _tagCloudBuildX23: {_tagTflite: 9.5},
+  _tagCloudBuildX28: {_tagTflite: 10},
+  _tagRyzen5600: {_tagTflite: 34},
+  _tagPixel5: {_tagTflite: 104},
+  _tagPixel6: {_tagPixelSo: 1015},
+  _tagIphone12mini: {
+    _tagTflite: 654,
+    _tagCoreml: 545,
+  },
+  _tagIosSimMBP2019: {_tagTflite: 23},
+};
+
+const Map<String, Map<String, double>> _objectDetection = {
+  _tagCloudBuildX23: {_tagTflite: 5.2},
+  _tagCloudBuildX28: {_tagTflite: 5.1},
+  _tagRyzen5600: {_tagTflite: 18},
+  _tagPixel5: {_tagTflite: 48},
+  _tagPixel6: {_tagPixelSo: 440},
+  _tagIphone12mini: {
+    _tagTflite: 259,
+    _tagCoreml: 305,
+  },
+  _tagIosSimMBP2019: {_tagTflite: 12.5},
+};
+
+const Map<String, Map<String, double>> _imageSegmentation = {
+  _tagCloudBuildX23: {_tagTflite: 2.0},
+  _tagCloudBuildX28: {_tagTflite: 1.9},
+  _tagRyzen5600: {_tagTflite: 6},
+  _tagPixel5: {_tagTflite: 30},
+  _tagPixel6: {_tagPixelSo: 160},
+  _tagIphone12mini: {
+    _tagTflite: 22,
+    _tagCoreml: 75,
+  },
+  _tagIosSimMBP2019: {_tagTflite: 4},
+};
+
+const Map<String, Map<String, double>> _naturalLanguageProcessing = {
+  _tagCloudBuildX23: {_tagTflite: 0.9},
+  _tagCloudBuildX28: {_tagTflite: 1},
+  _tagRyzen5600: {_tagTflite: 3},
+  _tagPixel5: {_tagTflite: 2.5},
+  _tagPixel6: {_tagPixelSo: 67},
+  _tagIphone12mini: {
+    _tagTflite: 12,
+    _tagCoreml: 81,
+  },
+  _tagIosSimMBP2019: {_tagTflite: 2.3},
+};
+
+const Map<String, Map<String, double>> _imageClassificationOffline = {
+  _tagCloudBuildX23: {_tagTflite: 12},
+  _tagCloudBuildX28: {_tagTflite: 12},
+  _tagRyzen5600: {_tagTflite: 52},
+  _tagPixel5: {_tagTflite: 152},
+  _tagPixel6: {_tagPixelSo: 1550},
+  _tagIphone12mini: {
+    _tagTflite: 1600,
+    _tagCoreml: 1250,
+  },
+  _tagIosSimMBP2019: {_tagTflite: 35},
+};
+
+const taskExpectedPerformance = {
+  'image_classification': _imageClassification,
+  'object_detection': _objectDetection,
+  'image_segmentation_v2': _imageSegmentation,
+  'natural_language_processing': _naturalLanguageProcessing,
+  'image_classification_offline': _imageClassificationOffline,
+};
+
+const expectedInstabilityMap = {
+  _tagCloudBuildX23: {_tagTflite: 1.15},
+  _tagCloudBuildX28: {_tagTflite: 1.15},
+  _tagRyzen5600: {_tagTflite: 1.05},
+  _tagPixel5: {_tagTflite: 1.15},
+  _tagPixel6: {_tagPixelSo: 1.1},
+};
```

Comment:
I think in this case it's better to use deviation than interval.
It's easier to look for issues this way.

Currently you can clearly see that we use 10% margin for the Pixel backend, but we need to use 15% for tflite.
With intervals you would have to calculate margins by hand.

The accuracy is different. Knowing that accuracy interval width is `0.10` would indicate that there is something wrong with a backend.

Reply:
I would actually suggest to do the opposite: use difference for accuracy, instead of just raw interval.
Except use absolute difference, instead of relative values.
It would look something like this:
```dart
const Map<String, Interval> _objectDetection = {
  'cpu': Interval(min: 0.31, maxDiff: 0.01),
  'npu': Interval(min: 0.28, maxDiff: 0.03),
  'tpu': Interval(min: 0.36, maxDiff: 0.02),
```

This way we would highlight the important values:
1) Min allowed value. Used to proof that backend works
2) How stable values are

Reply:
One thing that could be worth doing for throughput expected values is defining expected deviation for each task+backend+device individually, instead of only doing backend+device.

But in my experience a backend+device combination has pretty uniform stability of results across tasks.
So it's something to consider in the future, if we find out that on some devices one benchmark is significantly less stable that others.

Reply:
Using absolute or relative value for the tolerance, either of them is fine for me. It should only be consistent for both accuracy and throughput test.
I suggest differentiating between lower and upper bound for the deviation. Taking the idea from a [quality control chart](https://upload.wikimedia.org/wikipedia/commons/9/93/Xbar_chart_for_a_paired_xbar_and_R_chart.svg), we can define the expected value as:
```dart
// example for accuracy
ExpetedValue(mean: 0.8, percLowerThreshold: 0.10, percUpperThreshold: 0.15)
// example for throughput
ExpetedValue(mean: 34, percLowerThreshold: 0.25, percUpperThreshold: 0.5)
```
The class `ExpetedValue` can have convenient methods to calculate the absolute values: `absLowerThreshold()`, `absUpperThreshold()`.

Reply:
> It should only be consistent for both accuracy and throughput test.

Accuracy and throughput values are not consistent in its nature:
Accuracy is limited and normalized to `[0.0, 1.0]` interval.
Throughput is not limited, is not normalized, can not be normalized.

> I suggest differentiating between lower and upper bound for the deviation.

I don't see how we could use it in an integration test.

Quality control chart is built on a series of measurements.
An integration test doesn't have access to data from other tests (if we are speaking of CI).
It could be useful for analyzing how performance values change between many test runs, but we will need to use an external tool to do it.

Reply:
I'll change expected throughput values to specify allowed deviation for each value individually.

I think we shouldn't touch accuracy files in this PR. You can create a separate issue for it if you want.

Reply:
I have something else in mind, but here is not a right place for longer discussion, so let's finish this one first and update it later if needed.